test/functional/feature_assumeutxo.py

```diff
@@ -219,6 +216,19 @@ def test_snapshot_in_a_divergent_chain(self, dump_output_path):
         assert_equal(loaded['coins_loaded'], SNAPSHOT_BASE_HEIGHT)
         assert_equal(loaded['base_height'], SNAPSHOT_BASE_HEIGHT)
 
+        # Restart the node and delete the snapshot chainstate from previous test.
+        self.restart_node(2, extra_args=['-reindex-chainstate=1', *self.extra_args[2]])
+        assert node.getblockcount() < SNAPSHOT_BASE_HEIGHT
+
+        self.log.info(f"Check importing a snapshot where current chain-tip is not an ancestor or a descendant of the snapshot block and has more work")
```

Comment:
This may be resolved through addressing the comments on the previous test case anyway but for this one also, both chains should be valid.